payroll/search.py

```diff
@@ -100,50 +129,30 @@ class PayrollSearchMixin(object):
         'headcount'
     ]
 
-    def search(self, params, *args):
-
-        required_params = {
-            'name',
-            'employer',
-            'parent',
-            'taxonomy',
-            'universe',
-        }
-
-        if not set(params) & required_params:
-            # we want to allow a search for top paid employees
-            salary_above = params.get('salary_above')
-            if not salary_above or int(salary_above) < 150000:
-                raise DisallowedSearchException
-
-        if params.get('name') and len(params['name']) < 3:
-            raise DisallowedSearchException
-
+    def search(self, params, pagesize, **extra_kwargs):
         if params.get('entity_type'):
             entity_types = params.pop('entity_type').split(',')
         else:
             entity_types = ['unit', 'department', 'person']
 
         query_string = self._make_querystring(params)
 
-        if query_string:
-            for entity_type in entity_types:
-                yield from getattr(self, '_search_{}'.format(entity_type))(query_string, *args)
+        if params.get('page'):
+            offset = (int(params['page']) - 1) * pagesize
+            extra_kwargs.update({'start': offset})
```

Comment:
fascinating. can we also limit the result set since it's not going to be larger than page size?

Reply:
we do, down in L154.  but that can be moved up here, for sure. (i had to hunt for it when i came back to this code.)

Reply:
good idea. 

if we are doing this type of pagination here, then why do we need LazyPaginatedResults at all?

Reply:
need the count method so the pagination navigation shows up and works correctly.